cli/flox/src/commands/publish.rs

```diff
@@ -229,13 +200,18 @@ fn check_target_exists(lockfile: &Lockfile, package: &str) -> Result<bool> {
 }
 
 /// Merge cache values from config and args, with args taking precedence.
-/// Values must be mutually present or absent.
-fn merge_cache_options(
+/// If there's an ingress URI, there must be a signing key.
+async fn merge_cache_options(
+    flox: &Flox,
+    catalog_name: impl AsRef<str> + Send + Sync,
     config: Option<PublishConfig>,
     args: CacheArgs,
-    ingress_uri: Option<Url>,
 ) -> Result<Option<NixCopyCache>> {
-    let url = args.store_url.or(ingress_uri);
+    let url = match args.store_url {
+        Some(url) => Some(url),
+        // Get the ingress URI for this catalog if it has one configured.
+        None => flox.catalog_client.get_ingress_uri(catalog_name).await?,
```

Comment:
(non-blocking, on the understanding that this PR is prep and doesn't close the ticket) Do you have any thoughts on how we'll identify a catalog that hasn't been configured from the default state?

Reply:
I was loosely thinking that `get_ingress_uri` would return something like `Unconfigured | MetaOnly | Uri(String)`. Not completely sure how the openAPI typing will shake out